### PR TITLE
Configure using Offline Chat from Emacs

### LINKED ISSUE #358: Add offline chat support to Emacs
We added support for Llama V2 in Khoj with #352 , but it's still not accessible through Obsidian/Emacs. Add support for in-app experience.

Relates to #357

### PR DESCRIPTION
**Pull request by debanjum**
Allow enabling, disabling offline chat from Emacs during khoj setup

- Chat Options
  - Enable offline chat with `(setq khoj-chat-offline t)` during khoj setup
    - Offline chat models are better for privacy but not great at answering questions
  - Enable online chat by setting your OpenAI API key with `(setq khoj-openai-api-key <your-openai-api-key)`
    - Online chat models are better at answering but require internet

- Note: *Enabling offline chat will result in large download of Llama v2 7B model on your Khoj backend*

Related #330, #352, #359
Closes #358

### CODE REVIEW

File: src/interface/emacs/khoj.el [hunk 1]
```diff
@@ -428,40 +434,44 @@ CONFIG is json obtained from Khoj config API."
              (openai (assoc 'openai conversation)))
         (when openai
           ;; Unset the `openai' field in the khoj conversation processor config
-          (message "khoj.el: disable Khoj Chat using OpenAI as your OpenAI API key got removed from config")
+          (message "khoj.el: Disable Chat using OpenAI as your OpenAI API key got removed from config")
           (setcdr conversation (delq openai (cdr conversation)))
```
Comment (sabaimran) [inline src/interface/emacs/khoj.el:438]: Do you have to do anything to set the new value of `conversation` in `processor`? Or will this transfer automatically?

File: src/interface/emacs/khoj.el [hunk 2]
```diff
@@ -231,6 +231,11 @@ for example), set this to the full interpreter path."
   :type 'string
   :group 'khoj)
 
+(defcustom khoj-chat-offline nil
```
Comment (sabaimran) [inline src/interface/emacs/khoj.el:234]: Can you set a boolean for this field to `False` by default?

**Review by sabaimran — APPROVED** [review]
🏮

File: src/interface/emacs/khoj.el (hunk 2 above)
Comment (debanjum) [reply src/interface/emacs/khoj.el:234]: Yes, when `khoj-chat-offline` is `nil`, we set `enable-offline-chat` to `false` in `khoj.yml` 

**Background**:
- While elisp has a bi-state boolean (i.e nil in elisp equates to false). In the khoj config we support a 3 states even for booleans (null, true and false)
- The elisp to json convertors therefore use :false, :null to differentiate between the two possible meanings of nil in a tri-state scenario

File: src/interface/emacs/khoj.el (hunk 1 above)
Comment (sabaimran) [reply src/interface/emacs/khoj.el:438]: Nevermind, clarified offline.

Comment (debanjum) [reply src/interface/emacs/khoj.el:438]: We just set `api-key` field under processor > conversation > openai section to null in this section. The next section adds this updated conversation section to processor

**Pull request merged**